renku/cli/_checks/migrate_datasets.py

```diff
@@ -0,0 +1,226 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Checks needed to determine dataset migration policy."""
+import shutil
+import uuid
+from collections import defaultdict
+from pathlib import Path
+
+import click
+
+from renku.models.datasets import Dataset
+from renku.models.refs import LinkReference
+
+from .._echo import WARNING
+
+
+def dataset_pre_0_3(client):
+    """Return paths of dataset metadata for pre 0.3.4."""
+    return (client.path / 'data').rglob(client.METADATA)
+
+
+def check_dataset_metadata(client):
+    """Check location of dataset metadata."""
+    # Find pre 0.3.4 metadata files.
+    old_metadata = list(dataset_pre_0_3(client))
+
+    if not old_metadata:
+        return True
+
+    click.secho(
+        WARNING + 'There are metadata files in the old location.'
+        '\n  (use "renku migrate datasets" to move them)\n\n\t' + '\n\t'.join(
+            click.style(str(path.relative_to(client.path)), fg='yellow')
+            for path in old_metadata
+        ) + '\n'
+    )
+
+    return False
+
+
+def check_missing_files(client):
+    """Find missing files listed in datasets."""
+    missing = defaultdict(list)
+
+    for path, dataset in client.datasets.items():
+        for file_ in dataset.files:
+            filepath = Path(file_.path)
+            if not filepath.exists():
+                missing[str(
+                    path.parent.relative_to(client.renku_datasets_path)
+                )].append(str(filepath))
+
+    if not missing:
+        return True
+
+    click.secho(
+        WARNING + 'There are missing files in datasets.'
+        # '\n  (use "renku dataset clean <name>" to clean them)'
+    )
+
+    for dataset, files in missing.items():
+        click.secho(
+            '\n\t' + click.style(dataset, fg='yellow') + ':\n\t  ' +
+            '\n\t  '.join(click.style(path, fg='red') for path in files)
+        )
+
+    return False
+
+
+def check_dataset_resources(client):
+    """Find missing datasets or files listed in metadata."""
+    missing_datasets = defaultdict(list)
+    missing_files = defaultdict(list)
+
+    for path, dataset in client.datasets.items():
+
+        if not (Path(dataset.path) / Path(client.METADATA)).exists():
+            missing_datasets[path] = dataset
+
+        for file_ in dataset.files:
+            filepath = Path(file_.path)
+            if not filepath.exists():
+                relative = path.parent.relative_to(client.renku_datasets_path)
+                missing_files[str(relative)].append(str(filepath))
+
+    return missing_datasets, missing_files
+
+
+def ensure_clean_lock(client):
+    """Make sure Renku lock file is not part of repository."""
+    lock_file = Path('.renku.lock')
+    if lock_file.exists():
+        lock_file.unlink()
+
```

Comment:
If the lock file is deleted, it doesn't get properly comitted without: 
```suggestion
client.repo.git.add(str(lock_file))

```

Reply:
Could you please describe the steps on how I can reproduce the issue?